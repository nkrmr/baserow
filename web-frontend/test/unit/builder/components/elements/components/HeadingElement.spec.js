import { TestApp } from '@baserow/test/helpers/testApp'
import HeadingElement from '@baserow/modules/builder/components/elements/components/HeadingElement.vue'

describe('HeadingElement', () => {
  let testApp = null
  let store = null

  beforeAll(() => {
    testApp = new TestApp()
    store = testApp.store
    store.$registry.registerNamespace('element')
  })

  afterEach(() => {
    testApp.afterEach()
  })

  const mountComponent = ({ props = {}, slots = {}, provide = {} }) => {
    return testApp.mount(HeadingElement, {
      propsData: props,
      slots,
      provide,
    })
  }

  test('Default HeadingElement component', async () => {
    const builder = { id: 1, theme: { primary_color: '#ccc' } }
    const page = {}
    const workspace = {}
    const element = { level: 2, value: '', styles: {} }
    const mode = 'public'

    const wrapper = await mountComponent({
      props: {
        element,
      },
      provide: {
        builder,
        page,
        mode,
        applicationContext: { builder, page, mode },
        element,
        workspace,
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('Default HeadingElement component', async () => {
    const builder = { id: 1, theme: { primary_color: '#ccc' } }
    const page = {}
    const workspace = {}
    const element = { level: 3, value: '"hello"', styles: {} }
    const mode = 'public'

    const wrapper = await mountComponent({
      props: {
        element,
      },
      provide: {
        builder,
        page,
        mode,
        applicationContext: { builder, page, mode },
        element,
        workspace,
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
