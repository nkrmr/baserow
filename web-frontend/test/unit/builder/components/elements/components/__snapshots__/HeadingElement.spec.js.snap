// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingElement Default HeadingElement component 1`] = `
<div
  class="element--no-value"
>
  <h2
    class="ab-heading ab-heading--h2"
  >
    
    headingElement.noValue
  
  </h2>
</div>
`;

exports[`HeadingElement Default HeadingElement component 2`] = `
<div
  class=""
>
  <h3
    class="ab-heading ab-heading--h3"
  >
    
    hello
  
  </h3>
</div>
`;
